pkg/metrics/client_go_adapter.go

```diff
@@ -63,9 +75,13 @@ func registerClientMetrics() {
 // (which isn't anywhere in an easily-importable place).
 
 type resultAdapter struct {
-	metric *prometheus.CounterVec
+	metric       *prometheus.CounterVec
+	customMetric clientmetrics.ResultMetric
 }
 
-func (r *resultAdapter) Increment(_ context.Context, code, method, host string) {
+func (r *resultAdapter) Increment(ctx context.Context, code, method, host string) {
 	r.metric.WithLabelValues(code, method, host).Inc()
+	if r.customMetric != nil {
+		r.customMetric.Increment(ctx, code, method, host)
```

Comment:
Sorry, I do not understand this. You both want to register them yourself and plug your own functionality into the existing adapter? Why isn't one of the two enough?

Reply:
Initially, the goal was to support custom requestResult metrics and keep the original rest_client_requests_total metrics. I changed the resultAdapter to the default value of `RegisterOpts.RequestResult` in 639312c0e24c606365d41c98d95d163913a209fc